ZFSin/zfs/module/zfs/vdev_file.c

```diff
@@ -261,21 +278,34 @@ vdev_file_open(vdev_t* vd, uint64_t* psize, uint64_t* max_psize,
 		0
 	);
 	dprintf("%s: set Sparse 0x%x.\n", __func__, status);
-#else // !KERNEL
-
-	vf->vf_handle = CreateFile(FileName,
-		GENERIC_READ | GENERIC_WRITE,
-		FILE_SHARE_READ /*| FILE_SHARE_WRITE*/,
-		NULL,
-		OPEN_EXISTING,
-		FILE_ATTRIBUTE_NORMAL /*| FILE_FLAG_OVERLAPPED*/,
-		NULL);
-	if (vf->vf_handle == INVALID_HANDLE_VALUE) {
+#else // !_KERNEL
+
+	zfs_file_t fp;
+	zfs_file_attr_t zfa;
+	int err;
+	FileName = vd->vdev_physpath;
+
+	if (FileName && FileName[0] == '#') {
+		uint8_t *end;
+		end = &FileName[0];
+		while (end && end[0] == '#') end++;
+		ddi_strtoull(end, &end, 10, &vd->vdev_win_offset);
+		while (end && end[0] == '#') end++;
+		ddi_strtoull(end, &end, 10, &vd->vdev_win_length);
+		while (end && end[0] == '#') end++;
+
+		FileName = end;
+	} else
+		FileName = vd->vdev_path;
+
+	err = zfs_file_open(FileName, vdev_file_open_mode(spa_mode(vd->vdev_spa)), 0, &fp);
+	if (err) {
```

Comment:
OK, adding the partition/len hack to vdev_file so zdb can read disks right?

Reply:
We have checked this partition/len hack in the above `if` block and tested it also.